examples/regression/Ridge2FoldCVRegularization.py

```diff
@@ -0,0 +1,374 @@
+# more costly when conducting regularized regression, wherein we need to
+# optimise the regularization parameter in a cross-validation (CV) scheme,
+# thereby recomputing the inverse for each parameter.  scikit-learn offers an
+# efficient leave-one-out CV (LOO CV) for its ridge regression which avoids
+# these repeated computations [loocv]_. Because we needed an efficient ridge that works
+# in predicting  for the reconstruction measures in :py:mod:`skmatter.metrics`
+# we implemented with :class:`skmatter.linear_model.RidgeRegression2FoldCV` an
+# efficient 2-fold CV ridge regression that uses a singular value decomposition
+# (SVD) to reuse it for all regularization parameters :math:`\lambda`. Assuming
+# we have the standard regression problem optimizing the weight matrix in
+#
+# .. math::
+#
+#      \begin{align}
+#         \|\mathbf{X}\mathbf{W} - \mathbf{Y}\|
+#      \end{align}
+#
+# Here :math:`\mathbf{Y}` can be seen also a matrix as it is in the case of
+# multi target learning. Then in 2-fold cross validation we would predict first
+# the targets of fold 2 using fold 1 to estimate the weight matrix and vice
+# versa. Using SVD the scheme estimation on fold 1 looks like this.
+#
+# .. math::
+#
+#      \begin{align}
+#          &\mathbf{X}_1 = \mathbf{U}_1\mathbf{S}_1\mathbf{V}_1^T,
+#                \qquad\qquad\qquad\quad
+#                \textrm{feature matrix }\mathbf{X}\textrm{ for fold 1} \\
+#          &\mathbf{W}_1(\lambda) = \mathbf{V}_1
+#                 \tilde{\mathbf{S}}_1(\lambda)^{-1} \mathbf{U}_1^T \mathbf{Y}_1,
+#                 \qquad
+#                 \textrm{weight matrix fitted on fold 1}\\
+#          &\tilde{\mathbf{Y}}_2 = \mathbf{X}_2 \mathbf{W}_1,
+#                 \qquad\qquad\qquad\qquad
+#                 \textrm{ prediction of }\mathbf{Y}\textrm{ for fold 2}
+#      \end{align}
+#
+# The efficient 2-fold scheme in `RidgeRegression2FoldCV` reuses the matrices
+#
+# .. math::
+#
+#      \begin{align}
+#          &\mathbf{A}_1 = \mathbf{X}_2 \mathbf{V}_1, \quad
+#           \mathbf{B}_1 = \mathbf{U}_1^T \mathbf{Y}_1.
+#      \end{align}
+#
+# for each fold to not recompute the SVD. The computational complexity
+# after the initial SVD is thereby reduced to that of matrix multiplications.
+
+
+# %%
+# We first create an artificial dataset to control the effective rank of the
+# feature matrix.
+
+effective_rank = 10
+
+X, y = make_regression(
+    n_samples=1000,
+    n_features=400,
+    n_informative=400,
+    effective_rank=effective_rank,
+    tail_strength=1e-9,
+    random_state=SEED,
+)
+
+
+singular_values = np.linalg.svd(X, full_matrices=False)[1]
+
+# %%
+
+
+plt.loglog(singular_values)
+plt.title("Singular values of our feature matrix X")
+plt.axvline(effective_rank, label="effective rank", color="black")
+plt.xlabel("index feature")
+plt.ylabel("singular value")
+plt.legend()
+plt.show()
+
+# %%
+# One can see the jump in the singular value distribution. That will be later
```

Comment:
really not sure what "jump" I'm supposed to look at. is it the drop around feature index 50? And if so, what does that signify?